src/NuGet.Core/NuGet.Commands/TrustedSignersCommand/TrustedSignerActionsProvider.cs

```diff
@@ -0,0 +1,303 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Packaging;
+using NuGet.Packaging.Signing;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.Commands
+{
+    public sealed class TrustedSignerActionsProvider
+    {
+        private readonly ITrustedSignersProvider _trustedSignersProvider;
```

Comment:
Metentioned about this on another pr, i think you should pass it in constructor. You may pass it as null for production code and initilize to default value in the constructor itself.

Reply:
Why should we pass it in the constructor with a null default when it will only be used in tests? I think is cleaner to make it internal and have the test override the value, instead of giving the option to a user of this API to add its own copy of the type.

Reply:
its used in production as well, just that you're default initializing it when you actually use it. I find this pattern really confusing. Passing it in constructor makes it really simple and straight forward. Also then you don't need test only internal member.

It will also help if tomorrow you need to pass different SourceRepository...

Reply:
Passing it in the constructor makes it so that external users of this provider can then override the source repository which would not make sense for any scenario this provider is designed to.

The use of this property is to be able to test some scenarios of this provider with mock responses, not to modify the behavior. This doesn't mean testing internal-only members, but not exposing members that should not be exposed. By making this internal only, we can limit the usage of this property to what it is designed to do.

I have a strong opinion about not exposing unnecessary things. If we do not explicitly need something to be public/publicly set (i.e. from the constructor) then we should not expose it.

This pattern is used throughout our product to avoid exposing properties that are only needed in tests. Internal properties are useful to help tests limit the scope through mocks.